### PR TITLE
build: improved querying for the container ID

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @landism,

Please review the following commits I made in branch nicks/ch283/pod:

6c78fe816857c1baece1a6fff9c62cabbc46a7c9 (2018-09-18 17:18:17 -0400)
build: improved querying for the container ID

### CODE REVIEW

File: internal/build/container_resolver.go [hunk 1]
```diff
@@ -35,26 +68,29 @@ func (r *ContainerResolver) ContainerIDForPod(ctx context.Context, podName k8s.P
...
-	}
-
-	// On GKE, we expect there to be one real match and one spurious match -- a
-	// container running "/pause" (see: http://bit.ly/2BVtBXB); filter it out.
-	if len(containers) > 2 {
-		var ids []string
-		for _, c := range containers {
-			ids = append(ids, k8s.ContainerID(c.ID).ShortStr())
+		return "", containerNotFound{
+			Message: fmt.Sprintf("no containers found in pod: %s", podName),
```
Comment (maiamcc) [inline internal/build/container_resolver.go]: nit: no MATCHING containers found in pod

File: internal/synclet/server_adapter.go [hunk 2]
```diff
@@ -20,7 +22,17 @@ func NewGRPCServer(del *Synclet) *GRPCServer {
 var _ proto.SyncletServer = &GRPCServer{}
 
 func (s *GRPCServer) GetContainerIdForPod(ctx context.Context, req *proto.GetContainerIdForPodRequest) (*proto.GetContainerIdForPodReply, error) {
-	containerID, err := s.del.ContainerIDForPod(ctx, k8s.PodID(req.PodId))
+	name, err := reference.ParseNamed(req.ImageId)
+	if err != nil {
+		return nil, err
+	}
+
+	ref, ok := name.(reference.NamedTagged)
```
Comment (maiamcc) [inline internal/synclet/server_adapter.go]: can do this directly: https://github.com/windmilleng/tilt/blob/04f02ee6dc1ffd1407ce56857240e8d6619316cf/internal/k8s/image.go#L132

File: internal/build/container_resolver.go [hunk 3]
```diff
@@ -18,13 +23,41 @@ func NewContainerResolver(dcli DockerClient) *ContainerResolver {
 	return &ContainerResolver{dcli: dcli}
 }
 
-// containerIdForPod looks for the container ID associated with the pod.
-// Expects to find exactly one matching container -- if not, return error.
-// TODO: support multiple matching container IDs, i.e. restarting multiple containers per pod
-func (r *ContainerResolver) ContainerIDForPod(ctx context.Context, podName k8s.PodID) (k8s.ContainerID, error) {
+// containerIdForPod looks for the container ID associated with the pod and image ID
+func (r *ContainerResolver) ContainerIDForPod(ctx context.Context, podName k8s.PodID, image reference.NamedTagged) (k8s.ContainerID, error) {
```
Comment (maiamcc) [inline internal/build/container_resolver.go]: niiiiice! we may want to rename this family of funcs to indicate that we're matching on image as well? or not, meh!

File: internal/synclet/server_adapter.go (hunk 2 above)
Comment (nicks) [reply internal/synclet/server_adapter.go]: nice!

File: internal/build/container_resolver.go (hunk 1 above)
Comment (nicks) [reply internal/build/container_resolver.go]: done

**Pull request merged**